GVFS/GVFS.UnitTests/Tracing/EventListenerTests.cs

```diff
@@ -8,47 +9,24 @@ namespace GVFS.UnitTests.Tracing
     public class EventListenerTests
     {
         [TestCase]
-        public void EventListener_TryRecordMessage_ExceptionThrownInternally_ReturnsFalseAndErrorMessage()
+        public void EventListener_RecordMessage_ExceptionThrownInternally_RaisesFailureEventWithErrorMessage()
         {
-            string expectedErrorMessage = "test error message";
+            string expectedErrorMessage = $"test error message unique={Guid.NewGuid():N}";
 
             TraceEventMessage message = new TraceEventMessage { Level = EventLevel.Error, Keywords = Keywords.None };
             TestEventListener listener = new TestEventListener(EventLevel.Informational, Keywords.Any)
             {
                 RecordMessageInternalCallback = _ => throw new Exception(expectedErrorMessage)
             };
 
-            string actualErrorMessage;
-            bool? actualResult = listener.TryRecordMessage(message, out actualErrorMessage);
+            Moq.Mock<IEventListenerEventSink> eventSink = new Moq.Mock<IEventListenerEventSink>();

```

Comment:
Don't need the `Moq.` since there is a `using`.  Also need to pass in `MockBehavior.Strict` with at `Setup` and call `VerifyAll` at the end.  This should be for all the `Mock` objects unless there is some reason not to follow those standards?

Reply:
> Don't need the Moq. since there is a using.
Yep!

> Also need to pass in MockBehavior.Strict
Yep!

> with at Setup and call VerifyAll at the end. This should be for all the Mock objects unless there is some reason not to follow those standards?
We'd like to verify the number of times these events were raised (to ensure we only get events raised when the state _changed_) which is not possible using the `Setup && VerifyAll` pattern, only with the `Verify(expr, times)` call.